### PR TITLE
Re-organizing secret store related articles

### LINKED ISSUE #956: How To: Retrieve a secret - Nonexistent file
Hi,

_Disclaimer: I could not find any documentation on how to format these issues, please forgive me if I made any mistakes, or did not adhere to contributing guidelines._

In this [section](https://docs.dapr.io/developing-applications/building-blocks/secrets/howto-secrets/#setting-up-a-secret-store-component), the documentation states as follows:

```To deploy in Kubernetes, save the file above to aws_secret_manager.yaml and then run:```

However, the referred _file above_ does not exist in the actual documentation page. I would fix the issue my self, however, I do not know what the contents of the file should be.

### PR DESCRIPTION
**Pull request by orizohar**
## Description

Re-organizing several pages that provide guidance on how to configure a secret store and how to call the secrets API.
- Guidance for configuring a component is now in one place (under operations->components) and includes an example
- API usage is not tabbed.
- Intelinking improved
- Consolidated overview contents in one place

## Issue reference

Closes #956

### CODE REVIEW

File: daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md [hunk 1]
```diff
@@ -6,42 +6,23 @@ weight: 2000
...
 
-Now that the secret store is set up, you can call Dapr to get the secrets for a given key for a specific secret store.
+Once you have a secret store set up, you can call Dapr to get the secrets for a given key for a specific secret store.
 
 For a full API reference, go [here](https://github.com/dapr/docs/blob/master/reference/api/secrets_api.md).
 
 Here are a few examples in different programming languages:
 
-### Go
+{{< tabs "Go" "Javascript" "Python" "Rust" "C#" >}}
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md:66]: Can we add a simple `curl` tab and example?

File: daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md [hunk 2]
```diff
@@ -6,42 +6,23 @@ weight: 2000
...
-```bash
-kubectl apply -f aws_secret_manager.yaml
-```
-
-To run locally, create a `components` dir containing the YAML file and provide the path to the `dapr run` command with the flag `--components-path`.
-
-Watch this [video](https://www.youtube.com/watch?v=OtbYCBt9C34&feature=youtu.be&t=1818) for an example on how to use the secrets API. Or watch this [video](https://www.youtube.com/watch?v=8W-iBDNvCUM&feature=youtu.be&t=1765) for an example on how to component scopes with secret components and the secrets API.
+Before retrieving secrets in your application's code, you must have a secret store component configured. See guidance on [how to configure a secret store]({{<ref secret-stores-overview>}}) and review [supported secret stores]({{< ref supported-secret-stores >}}) to see specific details required for different secret store solutions.
 
 ## Calling the secrets API
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md:15]: We should add a step here to create a new environment variable, then creating a enviroment variable secret store, then retrieve the secret

File: daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md [hunk 3]
```diff
@@ -6,42 +6,23 @@ weight: 2000
...
 
-Now that the secret store is set up, you can call Dapr to get the secrets for a given key for a specific secret store.
+Once you have a secret store set up, you can call Dapr to get the secrets for a given key for a specific secret store.
```
Comment (AaronCrawfis) [inline daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md]: We should add a step before here to create a new environment variable, then creating a enviroment variable secret store, then retrieve the secret

File: daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md (hunk 1 above)
Comment (orizohar) [reply daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md:66]: I've decided to add the `curl` example prior to this section. Keeping this as a reference for calling the API in your code

File: daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md (hunk 3 above)
Comment (orizohar) [reply daprdocs/content/en/developing-applications/building-blocks/secrets/howto-secrets.md]: Add setting up a secret store but used the JSON file option instead (I felt it is a clearer "store" emulation)

**Comment by orizohar** [comment]
@AaronCrawfis please see my additions to the file per your review.

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**